class/practice2.1/AdvancedHelloWorld.cpp

```diff
@@ -1,110 +1,94 @@
+        SaveData();
+        return returnValue;
     }
-    if (!found) {
-        if (argument2 == "delete") {
-            std::cout << "There is no such user " <<std::endl;
-            return 0;
-        }
-        std::cout << "Welcome " << WordToFind << std::endl;
-        std::ofstream fileEdit("database.txt", std::ios::app);
-        if (!fileEdit.is_open()) {
-            std::cerr << "Could not open the file for writing!" << std::endl;
-            return 1;
-        }
-        fileEdit << WordToFind << " 1" << std::endl;
-        fileEdit.close();
+public:
+    Database(const std::string& filePath) : database(filePath) {
+         LoadData();
     }
-    else {
-        if (argument2 == "delete") {
-
-            for (int i = 0; i < lines.size(); i++)
-            {
-                if (splitString(lines[i], ' ')[0] == WordToFind)
-                {
-                    lines.erase(lines.begin() + i);
-                }
+    void clearDatabase() {
+        statsData.clear();
+        SaveData();
+    }
+    std::string PerformChanges(const std::string& nameToFind, const std::string& additionalCommand) {
+        if (statsData.find(nameToFind)==statsData.end()) {
+            if (additionalCommand == deleteCommand) {
+               return "There is no such user ";
             }
-            std::ofstream fileEdit("database.txt", std::ios::trunc);
-            for (const auto& line : lines) {
-                if (line != "")
-                {
-                    fileEdit << line << std::endl;
-                }
+            return UserIncrement(nameToFind);
+        }
+        else {
+            if (additionalCommand == deleteCommand) {
+                return deleteUser(nameToFind);
             }
-            std::cout << "All data about this user was cleared" << std::endl;
-            return 0;
+            return UserIncrement(nameToFind);
         }
-
-        std::ofstream fileEdit("database.txt", std::ios::trunc);
-        std::vector<std::string> arraySplitted=splitString(lineFound, ' ');
-       int num = std::stoi(arraySplitted[1]);
-       num++;
-       lines[lineNumber - 1] = arraySplitted[0] +" " + std::to_string(num);
-       for (const auto& line : lines) {
-           if (line != "")
-           {
-               fileEdit << line << std::endl;
-           }
-       }
-       std::cout << "Hello again(x"<<num <<"),"<< WordToFind << std::endl;
-       fileEdit.close();
     }
+};
+int main(int argc, char* argv[])
+{
+    Database db("database.txt");
+    if (argc < 2) {
+        std::cout << "Please provide a name to find." << std::endl;
+        return 0;
+    }
+    std::string nameToFind;
+    nameToFind = argv[1];
+    const auto additionalCommand = std::string(argc > 2 ? argv[2] : "");
+    if (nameToFind == bread)
+    {
+        db.clearDatabase();
+        std::cout << "all data was cleared" << std::endl;
+        return 0;
+    }
+   std::cout << db.PerformChanges(nameToFind,additionalCommand)<< std::endl;
```

Comment:
move performChanges implementation here and then adjust the code. so the logic should look like heandling of bread command. if cmd => prepare data +> call db method => print some message => return

if it will be too long code then it may be wrapped to handler functions